Core/Graphics/CMakeLists.txt

```diff
@@ -44,3 +101,12 @@ target_link_to_dependencies(GraphicsInternal
     INTERFACE bgfx
     INTERFACE bimg
     INTERFACE bx)
+
+target_compile_definitions(GraphicsInternal
+    INTERFACE NOMINMAX)
+
+# compile as objc++
+if (APPLE)
```

Comment:
One thing of note is now that Graphics includes the Apple platform headers, we need to compile all the modules that consume Graphics as objc++. so far this has caused no functional or compilation related issues, but this seems significant.

@ryantrem, do you have any concerns about this change?

Reply:
Why do we need this flag? We use objective c++ in other places and didn't need to add any new flags, right?

Reply:
We have a few places where we will include objectivec++ headers, but as far as I see, we do this all in .mm source files, such as ViewController.mm for our MacOS playground, but my understanding is that when clang compiles a .mm file, it will compile it as objectivec++. If we want to include objectivec++ in other source, we'll have to add the flag.

Reply:
Is this because a header with objective c++ types is being included in a shared .cpp file or something?

Reply:
Yes, the MetalKit header is included in Graphics

Reply:
I was able to modify Graphics such that components that are including Graphics and GraphicsImpl.h are no longer consuming the platform headers via some clever use of templates thanks to @syntheticmagus. However GetNativeWindow now has an awkward way to invoke it where we use the window type as a template specialization. Alternatives were to remove this from the class outright and create some utility function to handle this, but this seems a bit awkward as well...